packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -139,112 +190,143 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
+            }
+          },
+        )}
+        role={role}
+        tabIndex={disabled || role === 'presentation' ? undefined : -1}
+        aria-selected={isSelected}
+        aria-haspopup={subMenuItems.length > 0 ? 'true' : undefined}
+        aria-controls={subMenuItems.length > 0 ? submenuId : undefined}
+        aria-expanded={subMenuItems.length > 0 ? popover.open : undefined}
+        aria-disabled={disabled}
+        {...(subMenuItems.length === 0
+          ? { ...handlers, ...rest }
+          : popover.getReferenceProps({ ...handlers, ...rest }))}
+      >
+        {startIcon && (
+          <ListItem.Icon as='span' aria-hidden>
+            {startIcon}
+          </ListItem.Icon>
+        )}
+        <ListItem.Content>
+          <div>{children}</div>
+          {sublabel && <ListItem.Description>{sublabel}</ListItem.Description>}
+        </ListItem.Content>
+        {!endIcon && subMenuItems.length > 0 && (
+          <ListItem.Icon as='span' aria-hidden>
+            <SvgCaretRightSmall />
+          </ListItem.Icon>
+        )}
+        {endIcon && (
+          <ListItem.Icon as='span' aria-hidden>
+            {endIcon}
+          </ListItem.Icon>
+        )}
+      </ListItem>
 
       {subMenuItems.length > 0 && popover.open && (
-        <Portal>
-          <MenuItemContext.Provider
-            value={{ ref: menuItemRef, setIsNestedSubmenuVisible }}
-          >
-            <Menu
-              setFocus={focusOnSubmenu}
-              ref={popover.refs.setFloating}
-              {...popover.getFloatingProps({
-                id: submenuId,
-                onPointerMove: () => {
-                  // pointer might move into a nested submenu and set isSubmenuVisible to false,
-                  // so we need to flip it back to true when pointer re-enters this submenu.
-                  setIsSubmenuVisible(true);
-                },
-              })}
+        <FloatingNode id={nodeId}>
+          <Portal>
+            <MenuItemContext.Provider
+              value={{
+                setActiveIndex,
+                listItemsRef,
+                setHasFocusedNodeInSubmenu,
+              }}
             >
-              {subMenuItems}
-            </Menu>
-          </MenuItemContext.Provider>
-        </Portal>
+              <Menu
+                setFocus={false}
+                ref={popover.refs.setFloating}
+                {...popover.getFloatingProps({
+                  id: submenuId,
+                  onPointerMove: () => {
+                    // pointer might move into a nested submenu and set isSubmenuVisible to false,
+                    // so we need to flip it back to true when pointer re-enters this submenu.
+                    setIsSubmenuVisible(true);
+                  },
+                })}
+              >
+                {subMenuItems.map((item, index) =>
+                  React.cloneElement(item, {
+                    key: index,
```

Comment:
You shouldn't need to set `key` here. `key` generally belongs to the user of the component.

Reply:
Ohh, I see, removed (https://github.com/iTwin/iTwinUI/pull/1942/commits/e85db420d91fa8db90562d4d3ac398a2d5389496).